DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+
+namespace DataStructures.Lists 
+{
+    public class CircularBuffer<T> 
```

Comment:
Can you please implement the `IEnumerable` interface? Please refer to how the Linear Collections were implemented, e.g.: List, Stack, Queue ... etc.

Reply:
I've Implemented both.